### ISSUE TITLE
Extension make Intelisense Slow

### ISSUE THREAD

**Issue by jnissen** [365businessdev/vscode-alxmldocumentation#75]
Hello, 
if i Have Installed this Extension Intelisense is very slow. 
In the Extecute Extension window is then Comment
Unresponsive

![ALXMLExtension](https://user-images.githubusercontent.com/7793064/144373332-834d94fc-e4ef-4f1b-8044-31c67ba07e18.png)
.

**Comment by sirhc101** [comment]
Hello @jnissen,

we're currently working on the new AL XML Documentation extension, version 2.0, which will be based on a new core. The new core currently looks much faster in our projects so we hope to solve this issue with the update as well.

Could u pls give me some additional infos? When does the unresponsive appear? While initialization or while editing? We recognized some unresponsive while editing large files (like "Sales-Post" Codeunit)

Thanks!
/chris

**Comment by jnissen** [comment]
Hello,
Installation works fine.
If the Extension ist installed All Intelisense Help are slow.....
If I press Strg+Space it´s appears "Loading" but nothing happens.... Also in Small Codeunits...
If it helps, I can test your Extension ....

**Comment by sirhc101** [comment]
Thanks for your update! Sure, I'll let this issue stay open, until it's fixed and will get back to you as soon as our new extension version is ready for testing.

**Comment by ZepposBE** [comment]
@sirhc101 ,
1st of all, ty for the utility provided by the extension.
I was wondering though, is this optimization still work in progress? Having the extension enabled is eating all my resources. Had to disable the extension to even be able to work properly. (No IntelliSense at all, delayed compilation of written code, ...)

**Comment by sirhc101** [comment]
@ZepposBE Yes, it's still a work in progress, but right now I have only little time to work on it.

**Comment by hhfiddelke** [comment]
I think there is a general design fail to use to much regex on the whole source file. I think performance would be much better, when reading file line by line and creating the structures while reading the file. 
I just used an awk - script to generate an AL- source file of 6 MB size it takes about one second to process, because it is reading the source line by line and creates internal structures, and at the end it writes the whole AL in one run.
If AL- XMLdocumentation is enabled, it crashes the AL- Extensionhost on every start.

**Comment by sirhc101** [comment]
Sorry, currently I do not find the time to work on the new version! 🙁 I hope that I will find time over Christmas and New Year to finish the new version. Then all open issues, including the performance issue will be fixed.
